### ISSUE TITLE
TypeError: style.getCiteProc is not a function

### ISSUE THREAD

**Issue by Jutdm** [dvanoni/notero#12]
Hello,

Thanks for this add-on ! I installed i(, but when trying to add a new document, this error message comes in :

**_"Failed to save item(s) to Notion. 
TypeError: style.getCiteProc is not a function"_**

Do you have an idea of what the problem could be, and how to solve it ?

Thank you !

**Comment by dvanoni** [comment]
Hi @Jutdm, thanks for reporting!

To help me track down the issue, could you open the Error Console in Zotero from **Tools > Developer > Error Console** and then try to save items into Notion again. Could you send me a screenshot of whatever messages you see there?

Also, what version of Zotero are you using?

**Comment by Jutdm** [comment]
Thank you for your quick response !

I'm on Zotero 5.0.

Here is what is written : 

![image](https://user-images.githubusercontent.com/94546519/142238199-5fd5a96a-33b8-4374-942b-6b99ea6bd5f9.png)

**Comment by dvanoni** [comment]
Hi @Jutdm, sorry for the delay getting back to you.

Can you let me know what format you have selected in Zotero's **Export** settings? (See screenshot below for what I'm referring to.) I believe the error you're running in to is related to the export format.

Also, can you let me know what other add-ons you're using (if any)?

Thanks!

![CleanShot 2021-12-11 at 17 22 33@2x](https://user-images.githubusercontent.com/299357/145695943-b6f829a3-8a50-44f8-b564-20cfdb30238d.png)

**Comment by Jutdm** [comment]
Hello @dvanoni, 

Thank you for your response ! 

I use a French format, and no other add-on. Here is the name of the format :

![image](https://user-images.githubusercontent.com/94546519/145707899-0c0447a3-03a7-4ce9-9879-3951499ec174.png)

Thank you,
Ju

**Comment by dvanoni** [comment]
@Jutdm thanks for this additional info! It's sounds like we're on the right track. I downloaded the format you're using to try to reproduce the error on my side, but I haven't been able to so far.

Would you mind downloading the latest version of the style to see if that helps at all? I wonder if there have been any fixes to it since you previously installed it. It looks like it was [last updated in February](https://www.zotero.org/styles?q=id%3Apresses-universitaires-de-rennes).

To additionally help with debugging, would you mind running a snippet of JavaScript in Zotero and sending me the output? You can open the window from **Tools > Developer > Run JavaScript** (or the French equivalent 😄). Here's the code I'd like you to run:

```js
let format = Zotero.Prefs.get('export.quickCopy.setting');
format = Zotero.QuickCopy.unserializeSetting(format);
Zotero.Styles.get(format.id);
```

It should output something like this:

<img width="900" alt="CleanShot 2021-12-12 at 10 05 55@2x" src="https://user-images.githubusercontent.com/299357/145722016-ef465482-9540-4c9d-b794-2a8f524f96d9.png">

Thanks!

**Comment by dvanoni** [comment]
Thanks @Jutdm! Unfortunately, I'm not seeing anything out of the ordinary in your screenshot. I was curious to see if `getCiteProc` was going to be returned as something unexpected, but as you can see, it appears to be a function just like in my screenshot...

I'm still digging in to this to see if I can figure out what's causing the issue. I'll keep you posted!

P.S. It looks like someone else has reported running in to the same error [in the Zotero Forums](https://forums.zotero.org/discussion/comment/394669/#Comment_394669).

**Comment by dvanoni** [comment]
@Jutdm would you mind doing me a favor and installing the test build posted above? It won't fix the issue, but it should enable some more error logging that will hopefully be visible from **Tools > Developer > Error Console**.

After installing the test build, could you open the Error Console and then try syncing items again to cause the error to occur? Then send me a screenshot of the error?

Thank you!

**Comment by Jutdm** [comment]
Hello !

I actually found a solution. I downloaded the "APA" format to show you the problem, and forgot to delete it after. I put the format "Presses Universitaires de Rennes" back as the default format, and the synchronization actually worked. If I delete APA, it doesn't, but I just have to keep it that way. 

I guess this solution is good enough, but thank you very much for your implication in this process ahah !



I just have another little question. If it doesn't take too much time, could you do something for this little issue, please ? 

When I add a new document, the title of the document is not added in the "Title" column. Instead, the quotation format of the name and date of the article is the title. 
For example : (Goldbeter-Merinfeld, 2021)  is the title of the document below, and I would like it to be "Patrimoine, argent et conflits familiaux. Introduction" instead.

![image](https://user-images.githubusercontent.com/94546519/146191516-97f5875b-14a4-4ec2-9cf9-6402cdeb68fd.png)

I would love to have it this way automatically : 
![image](https://user-images.githubusercontent.com/94546519/146191982-840eca72-91c3-4111-8b48-c16fcfc2d403.png)

I tried renaming the column "Title", but it doesn't work. I need to create an additionnal column called "Title", which is not the actual title of the document in the database. I don't know if I make sense here ahah.

Thank you very much !

**Comment by dvanoni** [comment]
Ah, I understand what's causing the issue now! Currently, the code [assumes that the APA style is available](https://github.com/dvanoni/notero/blob/030f9f6fa8247bdae51f08f05ca0e514782bb538/content/notero-item.ts#L59) and uses it to generate the name of the Notion page—the `(author, year)` text that you referred to. I hadn't considered that the APA style could be removed, so that was a bad assumption!

I think this issue is related to a couple others. Someone just mentioned in #14 that they'd also like to have the page name in Notion be the full article title, and others folks in #5 would like to use a different citation key as the page name. So, it sounds like the solution is to make this configurable and _not assume_ that a specific style is installed.

@Jutdm thank you so much for debugging this with me, and I'm glad you were able to find a workaround!

**Comment by Jutdm** [comment]
Thank you very much for your work, this pluggin is soooo helpful !

**Comment by swwwwws** [comment]
I get the same error when connecting Zotero with Obsidian. I followed the above steps, but it keeps showing the same error. 

<img width="1012" alt="image" src="https://user-images.githubusercontent.com/76928803/219104817-e0379163-e2f9-49db-8fc8-9467e0617f74.png">

<img width="752" alt="image" src="https://user-images.githubusercontent.com/76928803/219105142-4d247470-1b57-4a5c-9ab4-41512b16d006.png">

**Comment by dvanoni** [comment]
@swwwwws, just to confirm, do you have the `American Psychological Association 7th edition` style installed in Zotero? If not, can you install it from https://www.zotero.org/styles/apa and let me know if you're still encountering the issue?

**Comment by swwwwws** [comment]
@dvanoni thanks for the quick reply. Yes I installed APA style in Zotero. It was installed and to be sure I installed it again with the .csl file, before I posted yesterday. 

<img width="688" alt="image" src="https://user-images.githubusercontent.com/76928803/219414741-36078a25-15c1-4ab4-9f48-c00ff7b03932.png">

**Comment by dvanoni** [comment]
@swwwwws, I'm a bit stumped on this one. Would you mind sending me another screenshot of the Error Console? This time, could you open it and click the "Clear" button on the top right before triggering the error. Then once you've encountered the error, scroll to the bottom of the window (if necessary) to ensure you capture as much of the latest message as possible. Thanks!

**Comment by swwwwws** [comment]
@dvanoni This is the whole error log in the two pictures below. I cleared the log beforehand and then immediately triggered the error.

![image](https://user-images.githubusercontent.com/76928803/219603850-58bb3af9-045e-46b9-a56b-b0ac513e3309.png)

![image](https://user-images.githubusercontent.com/76928803/219603981-fd165a3c-3ac9-4999-a526-7fe83d8cfca1.png)

**Comment by dvanoni** [comment]
Thanks for the additional screenshots, @swwwwws. How are you triggering the error?

**Comment by swwwwws** [comment]
In Obsidian I us Command-P and select Zotero Better BibTex integration. Then the Zotero hovering search bar appears and I select a publication. After a short t time, I get the error in Obsidian.

![image](https://user-images.githubusercontent.com/76928803/219854141-17a9a629-f45d-42b5-970f-f54c5af018a9.png)

![image](https://user-images.githubusercontent.com/76928803/219854175-10799b72-b8a0-4057-9ac9-4ea9ff938714.png)

![image](https://user-images.githubusercontent.com/76928803/219854230-71e1563c-7c84-458e-b0bd-d569441ec427.png)

**Comment by dvanoni** [comment]
I did some digging, but I'm unfamiliar with the specifics of Better BibTeX and Obsidian Zotero Integration. I'd suggest reaching out to the maintainers of those plugins to see if they can dig into this further.

I'm not sure if I'm on the right path, but I found a [reference to a translator ID](https://github.com/mgmeyers/obsidian-zotero-integration/blob/5693ab27316278e11c5f10b6fc0d3501ebedc07a/src/bbt/getCiteKeyExport.ts#L5) in the code of Obsidian Zotero Integration, and the [location of that translator](https://github.com/retorquere/zotero-better-bibtex/blob/567bed9869a79f42f8e3ca5ae697896d518a44c0/translators/Better%20CSL%20JSON.json#L2) in Better BibTeX. You may be able to reference these when reaching out to those maintainers.

**Comment by swwwwws** [comment]
Thank you for your time and effort! I will bring it to the people who maintain it

**Comment by nicolnt** [comment]
Same as mentioned here by @Jutdm  https://github.com/dvanoni/notero/issues/12#issuecomment-994776511, I downloaded the APA style again and it worked, even if I already had a `apa.csl` in my Zotero `styles` folder.
Too bad I didn't copy it before updating to be able to compare both versions.

For me, the issue appeared through **Obsidian Zotero Integration** plugin. And contrary to what was mentioned here by @dufourc1  https://github.com/dvanoni/notero/issues/12#issuecomment-1489927978 I didn't have to create a citation format in **Obsidian Zotero Integration** plugin settings, updating apa.csl in Zotero `styles` folder solved it.

**Comment by dvanoni** [comment]
Thanks for sharing that thread, @nicolnt! I doubt the fix they mention will resolve the issue for Notero because Notero still relies on the APA style specifically, but I'm going to see if I can fix that by using a similar approach with a hard-coded format instead of APA.

**Comment by dvanoni** [comment]
Well, I don't know that I'm going to hear back from the Zotero developers on that thread. (I know they're very busy.)

I did some digging, and I think the code to use as a reference lives [here](https://github.com/zotero/zotero/blob/b15fb36f1b546d51b3908081f1fd2e71b73739f4/chrome/content/zotero/xpcom/editorInstance.js#L1622).

**Comment by dvanoni** [comment]
I think I may have finally fixed this! 🤞 [Version 0.6.2](https://github.com/dvanoni/notero/releases/tag/v0.6.2) no longer relies on the APA style being installed, so hopefully that resolves things.
